examples/winrm/windows_logical_disks.js

```diff
@@ -0,0 +1,249 @@
+var statusTypes = {
+    "0": "Ok",
+    "1": "Scan Needed",
+    "2": "Spot Fix Needed",
+    "3": "Full Repair Needed"
+}
+
+// Custom Driver Table to store Logical disk information
+var table = D.createTable(
+    "Logical Disks", 
+    [
+        { label: "Drive Letter", valueType: D.valueType.STRING },
+        { label: "Name", valueType: D.valueType.STRING },
+        { label: "Physical Disk", valueType: D.valueType.STRING },
+        { label: "Usage", unit: "%", valueType: D.valueType.NUMBER },
+        { label: "Free Space", unit: "GiB", valueType: D.valueType.NUMBER },
+        { label: "Size", unit: "GiB", valueType: D.valueType.NUMBER },
+        { label: "File System", valueType: D.valueType.STRING },
+        { label: "Type", valueType: D.valueType.STRING },
+        { label: "Is Readonly", valueType: D.valueType.STRING },
+        { label: "BitLocker Status", valueType: D.valueType.STRING },
+        { label: "BitLocker Protection Status", valueType: D.valueType.STRING },
+        { label: "BitLocker Encryption Pct", unit: "%", valueType: D.valueType.NUMBER },
+        { label: "Status", valueType: D.valueType.STRING }
+    ]
+);
+
+// Check for Errors on the WinRM command response
+function checkWinRmError(err) {
+    if (err.message) console.error(err.message);
+    if (err.code == 401) D.failure(D.errorType.AUTHENTICATION_ERROR);
+    if (err.code == 404) D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+    console.error(err);
+    D.failure(D.errorType.GENERIC_ERROR);
+}
+
+// Execute WinRM command
+function executeWinrmCommand(command) {
+    var d = D.q.defer();
+    winrmConfig.command = command;
+    D.device.sendWinRMCommand(winrmConfig, function (output) {
+        if (output.error === null) {
+            d.resolve(output);
+        } else {
+            checkWinRmError(output.error);
+        }            
+    });
+    return d.promise;
+}
+
+/**
+* @remote_procedure
+* @label Validate WinRM connectivity with the device
+* @documentation This procedure is used to validate the driver and credentials provided during association.
+*/
+function validate() {
+    D.q.all([
+        executeWinrmCommand(logicalDisksCmd),
+        executeWinrmCommand(bitLockerCmd)
+    ])
+        .then(function (results) {
+            results.forEach(function (output) {
+                if (output.error !== null) {
+                    console.error("Validation failed");
+                    return D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+                }
+            });
+            console.log("Validation successful");
+            return D.success();
+        })
+        .catch(function (err) {
+            console.error(err);
+            D.failure(D.errorType.GENERIC_ERROR);
+        });
+}
+
+/**
+* @remote_procedure
+* @label Get Logical Disks Info
+* @documentation This procedure retrieves information about logical disks installed on a Windows machine.
+*/
+function get_status() {
+    executeWinrmCommand(logicalDisksCmd)
```

Comment:
can we make a parallel call here between logicalDisksCmd and bitLockerCmd?

Reply:
Yes, we can make a parallel call between "logicalDisksCmd" and "bitLockerCmd".